src/modules/doctors/doctors.service.ts

```diff
@@ -11,27 +12,34 @@ export class DoctorsService {
   constructor(
     @InjectRepository(Doctor)
     private doctorsRepository: Repository<Doctor>,
-    @InjectRepository(User)
-    private usersRepository: Repository<User>,
-    @InjectRepository(Specialization)
-    private specializationsRepository: Repository<Specialization>,
+    private readonly usersService: UsersService,
+    private readonly specializationsService: SpecializationsService,
   ) {}
   async create(createDoctorInput: CreateDoctorInput): Promise<Doctor> {
-    const doctor = await this.doctorsRepository.create(createDoctorInput);
-    doctor.specialization = await this.specializationsRepository.findOne({
-      id: createDoctorInput.specializationId,
-    });
-    doctor.user = await this.usersRepository.findOne({
-      id: createDoctorInput.userId,
-    });
-
     try {
+      const doctor = this.doctorsRepository.create(createDoctorInput);
+      doctor.specialization = await this.specializationsService.findOneById(
+        createDoctorInput.specializationId,
+      );
+      doctor.user = await this.usersService.findOneById(
+        createDoctorInput.userId,
+      );
+
       return await this.doctorsRepository.save(doctor);
     } catch (err) {
       throw new HttpException(err, HttpStatus.BAD_REQUEST);
     }
   }
 
+  async delete(doctorId: string): Promise<boolean> {
+    try {
+      await this.doctorsRepository.delete(doctorId);
+      return true;
+    } catch (err) {
```

Comment:
I think you missed the conditions described in the task:
- The doctor can’t be removed if he is assigned to Room
- The doctor can’t be removed if he has a scheduled appointment

Reply:
It won't delete because of table relations, If needed I can create extra validation in this method

Reply:
@DenysLutskiy its better to create those validations because you need to know why exactly doctor can't be romoved, in your case you will just receive an erorr "FK_CONSTRAINT error" or something like this

Reply:
Agreed, in more complex system you may have a "response codes" more detailed than HTTP codes, for example 10021 - doctor has active room assignment; 10022 - doctor has a scheduled appointment.